### PR TITLE
Document change

### PR DESCRIPTION
**Pull request by marktwtn**
- Add threading model document
- Change the badge from travis CI to buildkite CI
- Change the description of GPU and OpenCL prerequisite

### CODE REVIEW

File: docs/build-n-test.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Building and Testing
 
 ## Prerequisites
-* You need to configure paths and flags of OpenCL installation in `mk/opencl.mk`.
 * Check JDK installation and set environment variable `JAVA_HOME` if you wish to specify.
 * If target platform lacks of Intel SSE instructions, multi-threaded pure C implementation would be used as the fallback.
+* Install the OpenCL and GPU driver before calculating the PoW with GPU.
 * For FPGA-based hardware accelerator, [Lampa Lab's Cyclone V FPGA PoW](https://github.com/LampaLab/iota_fpga) is taken as the basis.
      - File `soc_system.rbf` is only valid for DE10-nano board, and you have to synthesize to get appropriate `soc_system.rbf` for Arrow SoCKit board.
      - [RBF file](https://github.com/ajblane/dcurl/releases/tag/v1.0-SoCKit) can be downloaded from our release.
```
Comment (jserv) [inline docs/build-n-test.md]: @ajblane, Please prepare another link for RBF file hosted at https://github.com/dltcollab

File: docs/threading-model.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Threading model
+
+## Thread pool size
+The thread pool size initialization is written in the function `init_once()`.\
```
Comment (jserv) [inline docs/threading-model.md:5]: You shall describe how dcurl makes use of thread pool

Comment (marktwtn) [reply docs/threading-model.md:5]: I added some brief description of `dcurl` and the thread pool.

File: docs/build-n-test.md (hunk 1 above)
Comment (ajblane) [reply docs/build-n-test.md]: https://github.com/DLTcollab/iota_fpga/releases/tag/v0.3-sockit

Comment (marktwtn) [reply docs/build-n-test.md]: The link has been updated.

**Pull request merged**